src/LinearSolvers/preconditioners.jl

```diff
@@ -55,18 +79,28 @@ mutable struct BlockJacobiPreconditioner <: AbstractPreconditioner
         for (i,v) in enumerate(part)
             push!(partitions[v], i)
         end
-        Js = Vector{Matrix{Float64}}(undef, npart)
-        nJs = maximum(length.(partitions))
-        id = Matrix{Float64}(I, nJs, nJs)
+        lpartitions = VI(undef, npart)
```

Comment:
I would replace this line with broadcasting:
`lpartitions = length.(partitions)`